README.md

```diff
@@ -1,20 +1,88 @@
+|    R.2    | Rosetta API support with sub-account balances |         X        |         X         |         X        |         X         |    X    |      X     |      X      |        X       |        X       |        X       |         X         |                    |              |
+|    C.1    |   Cryptographic Verification of local state   |         X        |         X         |         X        |         X         |    X    |      X     |      X      |        X       |        X       |        X       |         X         |          X         |              |
+|    C.2    |  Cryptographic Proofs for remote state access |         X        |         X         |         X        |         X         |    X    |      X     |      X      |        X       |        X       |        X       |         X         |          X         |       X      |
+
+## Architecture
+
+### Components
+
+The Flow Data Provisioning Service is constituted of three main components.
+
+The **indexer** is at the core of the service. It indexes deltas into its own data structures and provides a low level
+API to expose its state.
+The **mapper** synchronizes the **indexer** with a source of deltas (`FilesystemChain` or `NetworkChain`.)
+The **chain** is in charge of fetching block sequence data (from a WAL in the filesystem for past sporks, or a network
+socket for live ones.)
 
-- index transaction events
-- events retrieval interface
-- REST API interface
-- DPS web client
+### Diagram
 
-## Interfaces
+The following diagram describes a simple overview of an example where the DPS is reading from two sporks. One of them
+is a live one, which sends
 
-1. Indexer: indexes changes/deltas per block
-2. Update Stream: gets all the trie updates per chunk
-3. Chain View: has block sequence data
\ No newline at end of file
+```text
+┌─────────────────┐
+│                 │
+│   Past Spork    │
+│                 │
+│  ┌───────────┐  │             ┌───────┐
+│  │           │  │             │       │
+│  │ Exec Node ├──┼────────────►│ WALog │
+│  │           │  │             │       │
+│  └───────────┘  │             └───┬───┘
+│                 │                 │
+│                 │                 │
+└─────────────────┘                 │
+                                    ▼
+                      ┌─────────────────────────────┐
+                      │  FSChain / LedgerWALMapper  │
+                      ├────────────▼▼▼──────────────┤
+                      │                             │                          ┌──────────┐
+                      │                             │  `GET /account/balance`  │          │
+                      │  DATA PROVISIONING SERVICE  │◄─────────────────────────┤  Client  │
+                      │          INDEXER            │                          │          │
+                      │                             │                          └──────────┘
+                      │                             │
```

Comment:
In terms of API, there would be three different ones:

1. Raw API: retrieve a value (`[]byte`) by key (`[]byte`) at a given height (`uint64`).
2. Ledger API: respond to a query (`*ledger.Query`) with a value (`ledger.Value`) at a given pathfinder version (`uint8`). The `ledger.Query` contains a state commitment (`ledger.State`) that can be mapped to the height.
3. Rosetta API: this doesn't need to go into the docs, it's fully documented on their own site: https://www.rosetta-api.org/docs/AccountApi.html and https://www.rosetta-api.org/docs/BlockApi.html in this case.

Reply:
So you think the diagram should show all three APIs represented? I can do that, but it might take quite some space.

Reply:
I'll leave it up to your judgement.